### PR TITLE
Implemented bindings for `ceil` timestamp operation

### LINKED ISSUE #8682: [FEA] Series datetime ceil
As a complement to `Series.dt.floor` (https://github.com/rapidsai/cudf/issues/7102), I'd like to be able to use call `ceil` to round the frequency level of the data to a specified frequency (noting that `ceil` implies "rounding up" to the desired frequency, while `floor` implies rounding down).

Based on comments in #7102 , this might require libcudf support.

```python
import pandas as pd
import numpy as np
​
np.random.seed(12)
​
s = pd.Series(pd.date_range('1/1/2018 11:59:00', periods=1000, freq='s')).sample(3)
s
518   2018-01-01 12:07:38
871   2018-01-01 12:13:31
797   2018-01-01 12:12:17
dtype: datetime64[ns]
```

```python
s.dt.ceil('H') # nearest hour (round up)
518   2018-01-01 13:00:00
871   2018-01-01 13:00:00
797   2018-01-01 13:00:00
dtype: datetime64[ns]
```

```python
s.dt.ceil('D') # nearest day (round up)
518   2018-01-02
871   2018-01-02
797   2018-01-02
dtype: datetime64[ns]
```

```python
s.dt.ceil('min') # nearest minute (round up)
518   2018-01-01 12:08:00
871   2018-01-01 12:14:00
797   2018-01-01 12:13:00
dtype: datetime64[ns]
```

### PR DESCRIPTION
**Pull request by shaneding**
Closes #8682. Added python bindings for `ceil` operation implemented in #8942.

### CODE REVIEW

File: python/cudf/cudf/core/series.py [hunk 1]
```diff
@@ -5923,6 +5923,13 @@ def _get_dt_field(self, field):
             data=out_column, index=self.series._index, name=self.series.name
         )
 
+    def _ceil_datetime(self, field):
```
Comment (shwina) [inline python/cudf/cudf/core/series.py:5926]: I think we want to expose this as the `.dt.ceil` method instead:

```python
In [16]: s = pd.Series(['2001-01-01-12:00:00', '2001-01-02-01:27:00', '2001-01-03-05:30:00'], dtype='datetime64[ns]')

In [17]: s
Out[17]:
0   2001-01-01 12:00:00
1   2001-01-02 01:27:00
2   2001-01-03 05:30:00
dtype: datetime64[ns]

In [18]: s.dt.ceil(freq='D')
Out[18]:
0   2001-01-02
1   2001-01-03
2   2001-01-04
dtype: datetime64[ns]

In [19]: s.dt.ceil(freq='H')
Out[19]:
0   2001-01-01 12:00:00
1   2001-01-02 02:00:00
2   2001-01-03 06:00:00
dtype: datetime64[ns]
```

File: python/cudf/cudf/tests/test_datetime.py [hunk 2]
```diff
@@ -1551,3 +1551,34 @@ def test_error_values():
...
+    ],
+)
+def test_ceil(data):
+    resolutions = ["D", "H", "T", "S", "L", "U", "N"]
+
+    for time_type in DATETIME_TYPES:
+        ps = pd.Series(data, dtype=time_type)
+        gs = cudf.from_pandas(ps)
+
+        for resolution in resolutions:
```
Comment (brandon-b-miller) [inline python/cudf/cudf/tests/test_datetime.py:1580]: my only nit would be that I think it's a little easier to debug if we just parameterize over all these things.

Comment (shaneding) [reply python/cudf/cudf/tests/test_datetime.py:1580]: so write individual testcases for each resolution?

File: python/cudf/cudf/core/column/datetime.py [hunk 3]
```diff
@@ -170,6 +170,9 @@ def values(self):
     def get_dt_field(self, field: str) -> ColumnBase:
         return libcudf.datetime.extract_datetime_component(self, field)
 
+    def ceil_datetime(self, field: str) -> ColumnBase:
```
Comment (shwina) [inline python/cudf/cudf/core/column/datetime.py:173]: Minor nit: can we not just call this `ceil`?

**Review by shwina — APPROVED** [review]

File: python/cudf/cudf/tests/test_datetime.py [hunk 4]
```diff
@@ -1551,3 +1551,31 @@ def test_error_values():
...
+                "2100-03-14 07:30:20",
+                "1970-01-01 00:00:09",
+                "1969-12-31 12:59:10",
+            ]
+        )
+    ],
+)
+@pytest.mark.parametrize("resolution", ["D", "H", "T", "S", "L", "U", "N"])
+def test_ceil(data, resolution):
+    for time_type in DATETIME_TYPES:
```
Comment (brandon-b-miller) [inline python/cudf/cudf/tests/test_datetime.py:1575]: this can also be parameterized

**Review by brandon-b-miller — APPROVED** [review]

**Comment by brandon-b-miller** [comment]
@gpucibot merge

**Pull request merged**